app/api/auth/headersMiddleware.ts

```diff
@@ -0,0 +1,12 @@
+import { Request, Response, NextFunction } from 'express';
+
+export default (req: Request, res: Response, next: NextFunction) => {
+  if (!['POST', 'DELETE', 'PUT', 'OPTIONS'].includes(req.method)) {
+    return next();
+  }
+  if (req.get('X-Requested-With') === 'XMLHttpRequest') {
+    return next();
+  }
+  res.status(401);
+  return res.json({ error: 'Unauthorized', message: 'X-Requested-With header was not found!' });
```

Comment:
maybe change the message to header was not sent on the request, to be more clear.
also match the error with what the new code (if changed) means

Reply:
done